content/blogs/2024-11-18-kestra-ion.md

```diff
@@ -0,0 +1,273 @@
+
+## Kestra's Internal Storage
+
+Kestra decided to adopt **ION format** for all the data stored in its internal storage. Regardless of the original format of the incoming data, Kestra transforms it into ION format before storing it.
+
+Amazon Ion is a richly-typed, self-describing, hierarchical data serialization format offering interchangeable binary and text representations. The ION format comes with multiple benefits, and you can read more about its advantages [here](https://amazon-ion.github.io/ion-docs/guides/why.html).
+
+By standardizing on the ION format for internal storage, Kestra ensures flexibility in handling any type of data. This approach eliminates the overhead of dealing with multiple data formats and helps standardize ETL (Extract, Transform, Load) and data processing across systems.
+
+## Standardized ETL
+
+Let's explore how Kestra's decision to store data in the ION format has helped standardize its ETL workflows. Consider a pipeline that pulls data from Snowflake, joins it with data from a MySQL table, and outputs the results as a CSV file to S3.
+
+As a prerequisite, we will push the orders data from the CSV file into Snowflake. Yon can do it using SQL commands in the Snowflake console or a simple Kestra flow as shown here:
+
+```yaml
+id: load_data_into_snowflake
+namespace: company.team
+
+tasks:
+  - id: download_csv
+    type: io.kestra.plugin.core.http.Download
+    uri: https://huggingface.co/datasets/kestra/datasets/raw/main/csv/orders.csv
+
+  - id: create_table
+    type: io.kestra.plugin.jdbc.snowflake.Query
+    url: jdbc:snowflake://<account_identifier>.snowflakecomputing.com
+    username: "{{ secret('SNOWFLAKE_USER') }}"
+    password: "{{ secret('SNOWFLAKE_PASSWORD') }}"
+    sql: |
+      CREATE TABLE IF NOT EXISTS my_db.my_schema.orders (
+        order_id INT,
+        customer_name STRING,
+        customer_email STRING,
+        product_id INT,
+        price DECIMAL,
+        quantity INT,
+        total DECIMAL
+      )
+
+  - id: load_data_to_stage
+    type: io.kestra.plugin.jdbc.snowflake.Upload
+    url: jdbc:snowflake://<account_identifier>.snowflakecomputing.com
+    username: "{{ secret('SNOWFLAKE_USER') }}"
+    password: "{{ secret('SNOWFLAKE_PASSWORD') }}"
+    from: '{{ outputs.download_csv.uri }}'
+    fileName: orders.csv
+    prefix: raw
+    stageName: "@my_db.my_schema.%orders"
+
+  - id: load_data_to_table
+    type: io.kestra.plugin.jdbc.snowflake.Query
+    url: jdbc:snowflake://<account_identifier>.snowflakecomputing.com
+    username: "{{ secret('SNOWFLAKE_USER') }}"
+    password: "{{ secret('SNOWFLAKE_PASSWORD') }}"
+    sql: |
+      COPY INTO my_db.my_schema.orders
+      FROM @my_db.my_schema.%orders
+      FILE_FORMAT = (TYPE = 'CSV' FIELD_OPTIONALLY_ENCLOSED_BY = '"' SKIP_HEADER = 1);
+```
+
+And we will upload the products from the CSV file into MySQL. Again, this can be done using SQL commands in any MySQL client or a simple Kestra flow as shown here:
```

Comment:
```suggestion
And we will upload the products from the CSV file into MySQL. Again, this can be done using SQL commands in any MySQL client or a short Kestra flow as shown here:
```